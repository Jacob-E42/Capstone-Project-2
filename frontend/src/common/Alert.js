import React, { useCallback, useContext, useState } from "react";
import { Alert as MuiAlert, Snackbar } from "@mui/material";
import AlertContext from "../context_providers/AlertContext";

function Alert({ msg, color = "success" }) {
	const { setMsg } = useContext(AlertContext);
	console.debug("Alert", "messages=", msg, "color=", color);

	// State to control the visibility of the alert message
	const [visible, setVisible] = useState(true);

	// Function to dismiss the alert message
	const onDismiss = useCallback(() => {
		setVisible(false);
		setMsg("");
	}, [setVisible, setMsg]);

	// return (
	// 	<Alert
	// 		// Set the color of the alert message
	// 		color={color}
	// 		// Specify if the alert is visible or hidden
	// 		isOpen={visible}
	// 		// Function to toggle the visibility of the alert
	// 		toggle={onDismiss}>
	// 		{msg}
	// 	</Alert>
	// );
	return (
		<Snackbar
			open={visible}
			autoHideDuration={6000}
			onClose={onDismiss}>
			<MuiAlert
				onClose={onDismiss}
				severity={color}
				sx={{ width: "100%" }}>
				{msg}
			</MuiAlert>
		</Snackbar>
	);
}

export default Alert;
