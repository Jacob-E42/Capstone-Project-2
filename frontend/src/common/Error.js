import React from "react";
import { Alert } from "reactstrap";

const Error = ({ msg, color }) => {
	return <Alert color={color}>{msg}</Alert>;
};

export default Error;
