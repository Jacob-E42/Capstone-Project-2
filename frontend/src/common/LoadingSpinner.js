import React from "react";
import { CircularProgress, Box } from "@mui/material";

function LoadingSpinner() {
	return (
		<Box sx={{ display: "flex", justifyContent: "center", p: 3 }}>
			<CircularProgress />
		</Box>
	);
}

export default LoadingSpinner;
