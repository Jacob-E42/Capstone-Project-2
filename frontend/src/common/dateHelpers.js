import {
	eachDayOfInterval,
	format,
	parseISO,
	subDays,
	getDay,
	differenceInCalendarDays,
	addDays,
	isToday
} from "date-fns";

// Return today's date in format "yyyy-mm-dd"
function getCurrentDate() {
	const currentDate = format(new Date(), "yyyy-MM-dd", { weekStartsOn: 0 });
	return currentDate;
}

//Return an array of Dates corresponding to the difference between the calendar day of dateString and the date
// that's num days before the dateString.
// The date passed in is included as the last element in the array. Therefore the number of returned elements is n + 1.
//
//dateString has format: "yyyy-mm-dd"
const getPreviousNumDays = (dateString, num) => {
	const date = parseISO(dateString);
	// console.log("date=", dateString, date);
	let days = eachDayOfInterval({ start: subDays(date, num), end: date });
	const daysStrings = days.map(date => {
		return format(date, "yyyy-MM-dd");
	});
	return daysStrings;
};
//Return an array of Dates corresponding to the difference between the calendar day of dateString and the date
// that's num days after the dateString.
//dateString has format: "yyyy-mm-dd"
const getNextNumDays = (dateString, num) => {
	if (num === 0) return [];
	const date = parseISO(dateString);
	let days = eachDayOfInterval({ start: addDays(date, 1), end: addDays(date, num) });
	const daysStrings = days.map(date => {
		return format(date, "yyyy-MM-dd");
	});

	return daysStrings;
};

//Return a string corresponding to the day of the week of the dateString
//dateString has format: "yyyy-mm-dd"
const getDayOfWeek = dateString => {
	const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
	const date = parseISO(dateString);

	// const day = days[getDay(date)];
	// console.log(day, getDay(date));
	return days[getDay(date)];
};

//get the range of dateStrings that the StreakSlider should render
// accepts the dateString of the day whose journal will be shown
function getDateRange(dateString) {
	const date = parseISO(dateString);
	const currentDate = parseISO(getCurrentDate());

	const difference = differenceInCalendarDays(currentDate, date);
	if (difference < 0) return new RangeError("The date cannot be in the future!");
	const prevDays = getPreviousNumDays(dateString, 30);
	const nextDays = difference <= 30 ? getNextNumDays(dateString, difference) : getNextNumDays(dateString, 30);

	const dateRange = [...prevDays, ...nextDays];

	// console.log("difference=", difference, "prevDays=", prevDays, "nextDays=", nextDays);
	return dateRange;
}

function getPastDate(dateString, num) {
	const currentDate = parseISO(dateString);
	let pastDate = subDays(currentDate, num);
	return format(pastDate, "yyyy-MM-dd");
}

function getFutureDate(dateString, num) {
	const currentDate = parseISO(dateString);
	let fututeDate = addDays(currentDate, num);
	return format(fututeDate, "yyyy-MM-dd");
}

function isCurrentDate(dateString) {
	const currentDate = parseISO(dateString);

	return isToday(currentDate);
}

export {
	getPreviousNumDays,
	getDayOfWeek,
	getCurrentDate,
	getNextNumDays,
	getDateRange,
	getPastDate,
	getFutureDate,
	isCurrentDate
};
