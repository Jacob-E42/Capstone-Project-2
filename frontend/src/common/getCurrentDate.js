function getCurrentDate() {
	const currentDate = new Date().toISOString().slice(0, 10); // e.g., "2023-07-25"
	return currentDate;
}

export default getCurrentDate;
