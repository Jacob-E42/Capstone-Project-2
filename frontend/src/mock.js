import React, { useCallback, useState } from "react";
import UserContext from "./context_providers/UserContext";
import AlertContext from "./context_providers/AlertContext";
import Request from "./api";
import ApiRequest from "./api";
import ApiContext from "./context_providers/ApiContext";
import JournalContext from "./context_providers/JournalContext";

const demoUser = {
	id: 11,
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	interests: ["baking", "jetskiing"]
};

const demoJournal = {
	id: 42,
	userId: 11,
	title: "Baking adventure",
	entryText:
		"Yesterday I tried baking for the first time like I've always wanted to. It was a complete disaster, but at least it was fun.",
	entryDate: "2023-07-24",
	emotions: null
};

const UserProvider = ({ children }) => {
	return <UserContext.Provider value={{ user: demoUser }}>{children}</UserContext.Provider>;
};

const AnonUserProvider = ({ children }) => {
	const [currentUser, setCurrentUser] = useState(null);

	const login = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.login(formData);
				if (token) {
					setCurrentUser(demoUser);
					return { success: true };
				} else throw new Error("Log in token in missing");
			} catch (errors) {
				console.error("login failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	const signup = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.signup(formData);
				console.log(token);
				if (token) setCurrentUser(demoUser);
				else throw new Error("missing sign in token");
				return { success: true };
			} catch (errors) {
				console.error("signup failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	return <UserContext.Provider value={{ currentUser, login, signup }}>{children}</UserContext.Provider>;
};

const AlertProvider = ({ children }) => {
	const [msg, setMsg] = useState("");
	const [color, setColor] = useState("primary");

	return <AlertContext.Provider value={{ msg, setMsg, color, setColor }}>{children}</AlertContext.Provider>;
};

const ApiProvider = ({ children }) => {
	const [request, setRequest] = useState(new ApiRequest());
	request.setToken(
		`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTEsImVtYWlsIjoidGVzdEB0ZXN0Lm5ldCIsImlhdCI6MTY5MjI5MDkxN30.8e12Cw-TMts0FmZALH61ch03kUMlW5QoCi60dlfhyWY`
	);

	return <ApiContext.Provider value={{ api: request }}>{children}</ApiContext.Provider>;
};

const JournalProvider = ({ children }) => {
	return <JournalContext.Provider value={{ journal: demoJournal }}>{children}</JournalContext.Provider>;
};

export { UserProvider, AnonUserProvider, AlertProvider, ApiProvider, JournalProvider };
