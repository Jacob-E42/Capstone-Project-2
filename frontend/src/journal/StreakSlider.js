import React from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import "./Slider.css";
import DayBlock from "./DayBlock";
import { getPreviousTenDays } from "../common/dateHelpers";

const StreakSlider = ({ date }) => {
	if (!date) throw new Error("StreakSlider is missing a date!");
	const settings = {
		accessability: true,
		arrows: true,
		dots: true,
		infinite: false,
		speed: 500,
		slidesToShow: 7,
		slidesToScroll: 3
	};

	const previousTenDays = getPreviousTenDays(date);
	const dayBlocks = previousTenDays.map((day, index) => {
		console.log(day);
		return (
			<DayBlock
				day={day}
				key={index}
			/>
		);
	});
	console.log(previousTenDays, dayBlocks);

	return (
		<div className="StreakSlider">
			<h2> Streak </h2>
			<Slider {...settings}>{dayBlocks}</Slider>
		</div>
	);
};

export default StreakSlider;
