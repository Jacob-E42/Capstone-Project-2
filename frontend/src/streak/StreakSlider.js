import React, { useEffect } from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import "./Slider.css";
import DayBlock from "./DayBlock";
import useLocalStorage from "../hooks/useLocalStorage";
import useValidateDate from "../hooks/useVerifyDate";
import { getCurrentDate, getDateRange } from "../common/dateHelpers";

import { useNavigate } from "react-router-dom";
import LoadingSpinner from "../common/LoadingSpinner";

const StreakSlider = ({ date }) => {
	const navigate = useNavigate();
	if (!useValidateDate(date)) navigate(`journal/${getCurrentDate()}`);
	const [datesToRender, setDatesToRender] = useLocalStorage("datesToRender", [date]);
	const [slidesLoaded, setSlidesLoaded] = useLocalStorage("slidesLoaded", false);

	useEffect(() => {
		setDatesToRender(getDateRange(date));
		setSlidesLoaded(true);
	}, [date, setDatesToRender, setSlidesLoaded]);

	const daySlides = datesToRender.map((day, index) => {
		return (
			<DayBlock
				day={day}
				key={index}
			/>
		);
	});
	console.log(datesToRender, datesToRender.length, typeof daySlides);

	if (!slidesLoaded) return <LoadingSpinner />;

	const totalSlides = datesToRender.length;

	const settings = {
		accessability: true,
		arrows: true,
		centerMode: false,
		className: "Slider",
		dots: true,
		infinite: false,
		initialSlide: datesToRender.length - 1,
		slide: "div",
		speed: 500,
		slidesToShow: process.env.NODE_ENV === "test" ? totalSlides : 10,
		slidesToScroll: 3
	};

	return (
		<div className="StreakSlider">
			<h2> Streak </h2>
			<Slider {...settings}>{daySlides}</Slider>
		</div>
	);
};

export default StreakSlider;
