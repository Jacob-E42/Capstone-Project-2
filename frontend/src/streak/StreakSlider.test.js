import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AlertProvider, UserProvider } from "../mock";
import StreakSlider from "./StreakSlider";
import { getCurrentDate } from "../common/dateHelpers";

test("StreakSlider renders without crashing", () => {
	render(
		<MemoryRouter>
			<UserProvider>
				<AlertProvider>
					<StreakSlider date={getCurrentDate()} />
				</AlertProvider>
			</UserProvider>
		</MemoryRouter>
	);
});

// test("StreakSlider matches snapshot", () => {
// 	const { asFragment } = render(
// 		<MemoryRouter>
// 			<UserProvider>
// 				<AlertProvider>
// 					<StreakSlider />
// 				</AlertProvider>
// 			</UserProvider>
// 		</MemoryRouter>
// 	);
// 	expect(asFragment()).toMatchSnapshot();
// });
