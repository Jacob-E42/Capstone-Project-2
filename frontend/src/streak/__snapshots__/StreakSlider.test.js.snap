// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreakSlider matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="StreakSlider"
  >
    <h2>
       Streak 
    </h2>
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <button
        class="slick-arrow slick-prev slick-disabled"
        data-role="none"
        style="display: block;"
        type="button"
      >
         Previous
      </button>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-15"
              >
                <p>
                  Tue
                </p>
                <p>
                  15
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-16"
              >
                <p>
                  Wed
                </p>
                <p>
                  16
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-17"
              >
                <p>
                  Thu
                </p>
                <p>
                  17
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-18"
              >
                <p>
                  Fri
                </p>
                <p>
                  18
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="4"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-19"
              >
                <p>
                  Sat
                </p>
                <p>
                  19
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="5"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-20"
              >
                <p>
                  Sun
                </p>
                <p>
                  20
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active"
            data-index="6"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-21"
              >
                <p>
                  Mon
                </p>
                <p>
                  21
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="7"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-22"
              >
                <p>
                  Tue
                </p>
                <p>
                  22
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="8"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-23"
              >
                <p>
                  Wed
                </p>
                <p>
                  23
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="9"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-24"
              >
                <p>
                  Thu
                </p>
                <p>
                  24
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="10"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-25"
              >
                <p>
                  Fri
                </p>
                <p>
                  25
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="11"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-26"
              >
                <p>
                  Sat
                </p>
                <p>
                  26
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="12"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-27"
              >
                <p>
                  Sun
                </p>
                <p>
                  27
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="13"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-28"
              >
                <p>
                  Mon
                </p>
                <p>
                  28
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="14"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-29"
              >
                <p>
                  Tue
                </p>
                <p>
                  29
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="15"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-30"
              >
                <p>
                  Wed
                </p>
                <p>
                  30
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="16"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-08-31"
              >
                <p>
                  Thu
                </p>
                <p>
                  31
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="17"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-01"
              >
                <p>
                  Fri
                </p>
                <p>
                  01
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="18"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-02"
              >
                <p>
                  Sat
                </p>
                <p>
                  02
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="19"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-03"
              >
                <p>
                  Sun
                </p>
                <p>
                  03
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="20"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-04"
              >
                <p>
                  Mon
                </p>
                <p>
                  04
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="21"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-05"
              >
                <p>
                  Tue
                </p>
                <p>
                  05
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="22"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-06"
              >
                <p>
                  Wed
                </p>
                <p>
                  06
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="23"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-07"
              >
                <p>
                  Thu
                </p>
                <p>
                  07
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="24"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-08"
              >
                <p>
                  Fri
                </p>
                <p>
                  08
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="25"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-09"
              >
                <p>
                  Sat
                </p>
                <p>
                  09
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="26"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-10"
              >
                <p>
                  Sun
                </p>
                <p>
                  10
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="27"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-11"
              >
                <p>
                  Mon
                </p>
                <p>
                  11
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="28"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-12"
              >
                <p>
                  Tue
                </p>
                <p>
                  12
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="29"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-13"
              >
                <p>
                  Wed
                </p>
                <p>
                  13
                </p>
              </a>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="30"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <a
                class="DayBlock"
                href="/journal/2023-09-14"
              >
                <p>
                  Thu
                </p>
                <p>
                  14
                </p>
              </a>
            </div>
          </div>
        </div>
      </div>
      <button
        class="slick-arrow slick-next"
        data-role="none"
        style="display: block;"
        type="button"
      >
         Next
      </button>
      <ul
        class="slick-dots"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
        <li
          class=""
        >
          <button>
            5
          </button>
        </li>
        <li
          class=""
        >
          <button>
            6
          </button>
        </li>
        <li
          class=""
        >
          <button>
            7
          </button>
        </li>
        <li
          class=""
        >
          <button>
            8
          </button>
        </li>
        <li
          class=""
        >
          <button>
            9
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
